go/vt/sqlparser/token.go

```diff
@@ -48,7 +47,7 @@ type Tokenizer struct {
 	nesting              int
 	multi                bool
 	specialComment       *Tokenizer
-	specialPosOffset     int
+	partialIdentifier    []byte
```

Comment:
This is gross. Don't save state between between calls to Scan(). Scan() needs to return the next token value in all cases.